Formula/lighthouse.rb

```diff
@@ -0,0 +1,35 @@
+class Lighthouse < Formula
+  desc "Rust Ethereum 2.0 Client"
+  homepage "https://github.com/sigp/lighthouse"
+
+  url "https://github.com/sigp/lighthouse.git",
+      tag:      "v1.4.0",
+      revision: "3b600acdc5bf9726367c18277a22486573b8b457"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "rustup-init" => :build
+
+  # Update the Rust toolchain version whenever lighthouse is updated.
+  def rust_toolchain
+    "1.52.1"
+  end
+
+  # Update the version string whenever lighthouse is updated.
+  def lighthouse_version_string
+    "v1.4.0-3b600ac"
+  end
+
+  def install
+    # This will install a rust toolchain to be used with lighthouse.
+    system Formula["rustup-init"].bin/"rustup-init", "-qy", "--no-modify-path",
+           "--default-toolchain", rust_toolchain, "--profile", "minimal"
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
+    system "make"
+    bin.install "target/release/lighthouse"
+  end
+
+  test do
+    assert_match "Lighthouse #{lighthouse_version_string}", shell_output("#{bin}/lighthouse --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Got it I'll update the test. I saw some other formulae doing that and thought it's ok